rust/src/data_catalog/unity/mod.rs

```diff
@@ -75,17 +36,416 @@ pub enum GetTableError {
         /// Error description
         message: String,
     },
+
+    /// Unknown configuration key
+    #[error("Unknown configuration key: {0}")]
+    UnknownConfigKey(String),
+
+    /// Unknown configuration key
+    #[error("Missing configuration key: {0}")]
+    MissingConfiguration(String),
+
+    /// Unknown configuration key
+    #[error("Failed to get a credential from UnityCatalog client configuration.")]
+    MissingCredential,
+}
+
+impl From<UnityCatalogError> for DataCatalogError {
+    fn from(value: UnityCatalogError) -> Self {
+        match value {
+            UnityCatalogError::UnknownConfigKey(key) => DataCatalogError::UnknownConfigKey {
+                catalog: "Unity",
+                key,
+            },
+            _ => DataCatalogError::Generic {
+                catalog: "Unity",
+                source: Box::new(value),
+            },
+        }
+    }
 }
 
-impl From<reqwest_middleware::Error> for DataCatalogError {
-    fn from(value: reqwest_middleware::Error) -> Self {
-        value.into()
+/// Configuration options for unity catalog client
+pub enum UnityCatalogConfigKey {
+    /// Url of a Databricks workspace
+    ///
+    /// Supported keys:
+    /// - `unity_workspace_url`
+    /// - `databricks_workspace_url`
+    /// - `workspace_url`
+    WorkspaceUrl,
+
+    /// Access token to authorize API requests
+    ///
+    /// Supported keys:
+    /// - `unity_access_token`
+    /// - `databricks_access_token`
+    /// - `access_token`
+    AccessToken,
+
+    /// Service principal client id for authorizing requests
+    ///
+    /// Supported keys:
+    /// - `azure_client_id`
+    /// - `unity_client_id`
+    /// - `client_id`
+    ClientId,
+
+    /// Service principal client secret for authorizing requests
+    ///
+    /// Supported keys:
+    /// - `azure_client_secret`
+    /// - `unity_client_secret`
+    /// - `client_secret`
+    ClientSecret,
+
+    /// Tenant id used in oauth flows
+    ///
+    /// Supported keys:
+    /// - `azure_tenant_id`
+    /// - `unity_tenant_id`
+    /// - `tenant_id`
+    TenantId,
+}
+
+impl FromStr for UnityCatalogConfigKey {
+    type Err = DataCatalogError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "workspace_url" | "unity_workspace_url" | "databricks_workspace_url" => {
+                Ok(UnityCatalogConfigKey::WorkspaceUrl)
+            }
+            "access_token" | "unity_access_token" | "databricks_access_token" => {
+                Ok(UnityCatalogConfigKey::AccessToken)
+            }
```

Comment:
Seems like this should also match AZURE_TENANT_ID, AZURE_CLIENT_ID and AZURE_CLIENT_SECRET, right?
Btw, I wouldn't use the `unity_` alias for Azure tenant and client because it's not Databrick/Unity specific.

Reply:
I have been going back and forth on this and hope to experiment a bit here.

The main nuisance I encountered was that in scenarios where you are juggling many connections sometime we pick up things from the environment that aren't intended.

Generally speaking though yes, these are quite common and users would likely expect this to be picked up.